clang/lib/CIR/Dialect/IR/CIRDialect.cpp

```diff
@@ -1382,6 +1382,35 @@ void TernaryOp::build(OpBuilder &builder, OperationState &result, Value cond,
     result.addTypes(TypeRange{yield.getOperandTypes().front()});
 }
 
+//===----------------------------------------------------------------------===//
+// SelectOp
+//===----------------------------------------------------------------------===//
+
+LogicalResult SelectOp::verify() {
+  auto trueValueTy = getTrueValue().getType();
+  auto falseValueTy = getFalseValue().getType();
+  auto resultTy = getType();
+
+  // true_value, false_value, and the result must have the same type.
+  if (trueValueTy != falseValueTy)
+    return emitOpError()
+           << "true_value and false_value must have the same type";
```

Comment:
I think this can be covered in tablegen by using `AllTypesMatch`

Reply:
Awesome, first time to hear of this trait! Updated.